src/app/result/page.tsx

```diff
@@ -1,19 +1,38 @@
-import { ShareButton } from '@/app/button';
+'use client';
+
+import { ShareButton } from '@/components/ui/Button';
+import { useSearchParams } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import Loading from './loading';
+
+export default function Page() {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState('');
+  const battlePowerResult = useSearchParams();
+  const score = Number(battlePowerResult.get('score'));
+
+  const error = score === null || Number.isNaN(Number(score)) ? '適切な値ではありません' : '';
```

Comment:
勝手にサジェスチョンしてすみませんmm
こんな感じで直りそうです

```suggestion
  const [error, setError] = useState('');
  const battlePowerResult = useSearchParams();
  const score = Number(battlePowerResult.get('score'));

  if (score === null || Number.isNaN(score)) {
    setError('適切な値ではありません');
  }else {
    setError('');
  }
```